features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/BitmapPoolTest.kt

```diff
@@ -49,17 +49,30 @@ internal class BitmapPoolTest {
 
     @BeforeEach
     fun setup(forge: Forge) {
-        internalCache = LruCache(MAX_CACHE_MEMORY_SIZE_BYTES)
-        testedCache.setBackingCache(internalCache)
+        internalCache = object : LruCache<String, Bitmap>(MAX_CACHE_MEMORY_SIZE_BYTES) {
```

Comment:
what is the purpose of providing an implementation here and not a Mock ? I see that you are not doing anything in the `entryRemoved` method ?

Reply:
It's a slightly more accurate representation of the lrucache we're using. The default implementation calculates size by number of objects, and here we're calculating size by bitmap allocation the same as in the class. Regarding mocking, we discussed before to use the actual lrucache and not a mock in order to make use of the cache in the tests

Reply:
but if you provide this custom implementation in your tests this means that you are altering the class functionality ? This would be acceptable if you want to mock other methods in order to isolate one test case but if you do this for the whole test - suite how is this going to be relevant anymore ? You say here: `It's a slightly more accurate representation of the lrucache we're using` . Would this mean that you are changing your internal logic ?